src/cljs-lib/src/calva/repl/webview/core.cljs

```diff
@@ -0,0 +1,229 @@
+                                              :content code-theme})
+      (util/log-to-console
+       :error
+       "Cannot set code theme in output webview. There is no code theme set for the ColorThemeKind enum value of"
+       color-theme-kind))))
+
+(defn create-color-theme-change-listener
+  [{:keys [^js vscode/vscode] :as context}
+   {:keys [webview-panel]}]
+  (.. vscode -window
+      (onDidChangeActiveColorTheme
+       (fn [e]
+         (set-code-theme! context {:color-theme-kind (.. e -kind)
+                                   :webview-panel webview-panel})))))
+
+(defn add-subscriptions!
+  [{vscode-context :vscode/context
+    :as context}
+   {:keys [webview-panel]}]
+  (let [subscriptions [(create-color-theme-change-listener context {:webview-panel webview-panel})]]
+    (run! (fn [subscription]
+            (.. ^js vscode-context -subscriptions (push subscription)))
+          subscriptions)))
+
+(defn create-repl-output-webview-panel
+  [{:keys [vscode/vscode] :as context}]
+  (let [webview-panel (.. ^js vscode -window
+                          (createWebviewPanel
+                           "calva:repl-output"
+                           "REPL Output"
+                           #js {:preserveFocus true
+                                :viewColumn (.. ^js vscode -ViewColumn -Beside)}
+                           #js {:enableScripts true
+                                ;; If performance or memory consumption becomes a problem, we can use the setState
+                                ;; and getState to manually retain the context of the webview when it's hidden.
+                                ;; See https://code.visualstudio.com/api/extension-guides/webview#persistence
+                                :retainContextWhenHidden true
+                                :enableFindWidget true}))]
+    (.. ^js webview-panel (onDidDispose (fn [] (dispose-repl-output-webview-panel repl-output-webview-panel))))
+    (set-webview-html! context {:webview-panel webview-panel})
+    (add-subscriptions! context {:webview-panel webview-panel})
+    webview-panel))
+
+(defn ^:export show-repl-output-webview-panel
+  [preserve-focus?]
+  (let [context {:env/is-debug (:is-debug util/env)
+                 :vscode/vscode @util/vscode
+                 :vscode/context @util/vscode-context}
+        ^js webview-panel (or @repl-output-webview-panel
+                              (reset! repl-output-webview-panel (create-repl-output-webview-panel context)))
+        active-code-theme-kind (.. ^js @util/vscode -window -activeColorTheme -kind)]
+    (.. webview-panel (reveal nil preserve-focus?))
+    (set-code-theme! context {:color-theme-kind active-code-theme-kind
+                              :webview-panel webview-panel})))
+
+(def output-category->command-name
+  {"otherOut" "show-stdout"
+   "evalOut" "show-stdout"
+   "evalResults" "show-result"
+   "evalErr" "show-stdout"
+   "otherErr" "show-stdout"
+   "clojure" "show-result"})
```

Comment:
We currently use "show-stdout" for all non-code output, but we can easily change that to make different output types look more distinct. This is just a starting point.